task-graphs-v2.yaml

```diff
@@ -0,0 +1,370 @@
+      name:
+        type: string
+        description: >
+          The name of this graph, to appear in URLs.
+          Must be unique per-namespace.
+      readme:
+        type: string
+        description: Documentation for the task graph, in Markdown format.
+      license_id:
+        type: string
+        x-nullable: True
+        description: SPDX license identifier.
+      license_text:
+        type: string
+        x-nullable: True
+        description: Full text of the license.
+      tags:
+        type: array
+        x-nullable: False
+        description: Optional tags to classify the graph.
+        items:
+          type: string
+          description: Unaccented, lowercase, hyphenated Latin characters.
+      nodes:
+        type: array
+        items:
+          $ref: '#/definitions/RegisteredTaskGraphNode'
+        description: >
+          The structure of the graph, in the form of the nodes that make it up.
+          As with `TaskGraphLog`, nodes must topologically sorted, so that any
+          node appears after all the nodes it depends on.
+      # Potential way to deduplicate identical code blocks.
+      # Not in initial version to keep it simpler.
+      # executables:
+      #   type: object
+      #   description: >
+      #     A dictionary of ID to base64-encoded executable code block.
+      #     Used to deduplicate code blocks shared by multiple Nodes.
+
+  RegisteredTaskGraphNode:
+    description: >
+      Information about a single node within a registered task graph.
+      A single node represents one piece of data or a computational step;
+      either as an input value, a data source, or a computation that acts upon
+      earlier nodes. The structure parallels the existing
+      `TaskGraphNodeMetadata`.
+    type: object
+    properties:
+      client_node_id:
+        type: string
+        description: The client-generated UUID of the given graph node.
+      name:
+        type: string
+        x-nullable: true
+        description: >
+          A client-specified name for the node. If provided, this must be
+          unique.
+      depends_on:
+        type: array
+        items:
+          type: string
+        description: >
+          The client_node_uuid of each node that this node depends upon.
+          Used to define the structure of the graph.
+      array_node:
+        $ref: '#/definitions/UDFArrayDetails'
+      input_node:
+        $ref: '#/definitions/TGInputNodeData'
+      sql_node:
+        $ref: '#/definitions/TGSQLNodeData'
+      udf_node:
+        $ref: '#/definitions/TGUDFNodeData'
+
+  TGInputNodeData:
+    description: >
+      Specifies that a node is an “input value”, allowing for parameterized
+      task graphs. An input node may not depend upon any other nodes.
+    type: object
+    x-nullable: True
+    properties:
+      default_value:
+        $ref: '#/definitions/TGImmediateValue'
+      result_format:
```

Comment:
subtle: We have a `result_format` here and a `result_format` in `default_value` These must be the same otherwise we imply that the node will make a conversion. I suggest to remove this `result_format`